Feat/header userdata logout

pavel-sobchenko opened the pull request:
https://github.com/users/rasulovdev/projects/3/views/1?pane=issue&itemId=30484678

![Screenshot 2024-06-13 at 11 16 08](https://github.com/pavel-sobchenko/crtlx-freelance/assets/37002874/e30be228-be71-4a97-b3fc-b103af059cdb)

pavel-sobchenko committed:
- add action, selector, effect for getting user information
- add header component
- add avatar component

pavel-sobchenko committed:
fix: lint fix

pavel-sobchenko committed:
Merge branch 'develop' into feat/header_userdata_logout

# Conflicts:
#	src/app/layout/components/navigation-item/navigation-item.component.ts
#	src/app/layout/pages/main-dashboard-page/main-dashboard-page.component.html
#	src/app/layout/pages/main-dashboard-page/main-dashboard-page.component.ts

pavel-sobchenko committed:
fix: removed extra div tags from components

pavel-sobchenko committed:
fix: add flowbite library

rasulovdev commented on src/app/core/auth/types/user.ts:
missed `email` field

rasulovdev commented on src/app/core/auth/types/user.ts:
```diff
@@ -0,0 +1,15 @@
+  avatar: string
+  name: string
+  createdAt: Date
+  updatedAt: Date
```
These fields are not needed in this object

rasulovdev commented on src/app/core/auth/types/user.ts:
```diff
@@ -0,0 +1,15 @@
+  city: string
+  postCode: string
+  country: string
+}
```
I'd add a separate file for this type somewhere in `/core/types` because we'll need it in future in several different features

rasulovdev commented on src/app/features/auth/pages/login/login-page.component.ts:
```diff
@@ -32,7 +32,11 @@ export class LoginPageComponent {
+      this._store.dispatch(new GetUserProfile())
+
+      return void this._router.navigate(['/'])
+    }
```
This logic should not depend on login/logout. I'd put it somewhere more globally and make it depend on `AuthService.isAuthenticated`.

rasulovdev commented on src/app/shared/components/avatar/avatar.component.html:
```diff
@@ -0,0 +1,26 @@
+<ng-container *ngIf="user">
```
Component should not decide if it should be hidden or shown in its' entirety. It should be decided by the consumer

rasulovdev commented on src/app/shared/components/avatar/avatar.component.html:
Component name does not fully describe what the actual component does. Reconsider the naming and/or its' functionality.

rasulovdev commented on src/app/layout/components/header/header.component.ts:
too much to write

pavel-sobchenko committed:
fix: add Address type in a separated file
fix: split user profile and avatar components
fix: add a header as directive extension

rasulovdev commented on src/app/layout/pages/main-dashboard-page/main-dashboard-page.component.ts:
```diff
@@ -1,13 +1,30 @@
+
+  constructor(private readonly _store: Store) {
+    effect(() => {
+      if (this._isAuthenticatedSignal()) {
+        this._store.dispatch(new GetUserProfile())
+      }
```
```ts
if (!this._isAuthenticatedSignal()) return
this._store.dispatch(new GetUserProfile())
```

rasulovdev commented on src/app/shared/components/avatar/avatar.component.html:
`noAvatar` template does not support sizes

pavel-sobchenko committed:
fix: avatar empty user icon size adjust
